Sources/SnapshotTesting/AssertSnapshot.swift

```diff
@@ -37,98 +37,121 @@ public func assertSnapshot<A, B>(
+        optionalDiffable = b
+        tookSnapshot.fulfill()
+      }
+      let result = XCTWaiter.wait(for: [tookSnapshot], timeout: timeout)
+      switch result {
+      case .completed:
+        break
+      case .timedOut:
+        return "Exceeded timeout of \(timeout) seconds waiting for snapshot"
+      case .incorrectOrder, .invertedFulfillment, .interrupted:
+        return "Couldn't snapshot value"
+      }
 
-    guard let diffable = optionalDiffable else {
-      XCTFail("Couldn't snapshot value", file: file, line: line)
-      return
-    }
+      guard let diffable = optionalDiffable else {
+        return "Couldn't snapshot value"
+      }
 
-    guard !recording, fileManager.fileExists(atPath: snapshotFileUrl.path) else {
-      try strategy.diffable.to(diffable).write(to: snapshotFileUrl)
-      XCTFail("Recorded snapshot: …\n\n\"\(snapshotFileUrl.path)\"", file: file, line: line)
-      return
-    }
+      guard !recording, fileManager.fileExists(atPath: snapshotFileUrl.path) else {
+        try strategy.diffable.to(diffable).write(to: snapshotFileUrl)
+        return "Recorded snapshot: …\n\n\"\(snapshotFileUrl.path)\""
+      }
 
-    let data = try Data(contentsOf: snapshotFileUrl)
-    let reference = strategy.diffable.fro(data)
+      let data = try Data(contentsOf: snapshotFileUrl)
+      let reference = strategy.diffable.fro(data)
 
-    guard let (failure, attachments) = strategy.diffable.diff(reference, diffable) else {
-      return
-    }
+      guard let (failure, attachments) = strategy.diffable.diff(reference, diffable) else {
+        return nil
+      }
 
-    let artifactsUrl = URL(
-      fileURLWithPath: ProcessInfo.processInfo.environment["SNAPSHOT_ARTIFACTS"] ?? NSTemporaryDirectory()
-    )
-    let artifactsSubUrl = artifactsUrl.appendingPathComponent(fileName)
-    try fileManager.createDirectory(at: artifactsSubUrl, withIntermediateDirectories: true)
-    let failedSnapshotFileUrl = artifactsSubUrl.appendingPathComponent(snapshotFileUrl.lastPathComponent)
-    try strategy.diffable.to(diffable).write(to: failedSnapshotFileUrl)
-
-    if !attachments.isEmpty {
-      #if !os(Linux)
-      XCTContext.runActivity(named: "Attached Failure Diff") { activity in
-        attachments.forEach {
-          activity.add($0.rawValue)
+      let artifactsUrl = URL(
+        fileURLWithPath: ProcessInfo.processInfo.environment["SNAPSHOT_ARTIFACTS"] ?? NSTemporaryDirectory()
+      )
+      let artifactsSubUrl = artifactsUrl.appendingPathComponent(fileName)
+      try fileManager.createDirectory(at: artifactsSubUrl, withIntermediateDirectories: true)
+      let failedSnapshotFileUrl = artifactsSubUrl.appendingPathComponent(snapshotFileUrl.lastPathComponent)
+      try strategy.diffable.to(diffable).write(to: failedSnapshotFileUrl)
+
+      if !attachments.isEmpty {
+        #if !os(Linux)
+        XCTContext.runActivity(named: "Attached Failure Diff") { activity in
```

Comment:
Still coupled to XCTest via XCTContext and XCTWaiter, but basically all testing in Swift piggybacks off XCTest these days anyway. We can revisit decoupling entirely if/when other test libraries become popular.